drivers/digit7seg/digit7seg.c

```
@@ -0,0 +1,158 @@
+
+static void _set_pin_value(digit7seg_t *dev)
+{
+    const gpio_t pins[] =
+    {
+        PIN_A, PIN_B, PIN_C, PIN_D, PIN_E, PIN_F, PIN_G, PIN_DP
+    };
+
+    uint8_t mask = 0xFF;
+    uint8_t current_value = (uint8_t)(dev->value >> (dev->current_digit * 8) & mask);
+
+    for (int i = 0; i < BYTE_BITS; i++) {
+        if (current_value & (1 << i)) {
+            gpio_set(pins[i]);
+            DEBUG("PIN SET\n");
+        }
+        else {
+            gpio_clear(pins[i]);
+            DEBUG("PIN CLEAR\n");
+        }
+    }
+}
+
+static void _shift_display(void *arg, int chan)
+{
+    (void)chan;
+    digit7seg_t *dev = (digit7seg_t*)arg;
+
+    const gpio_t digit_pins[] =
+    {
+        PIN_DIG1, PIN_DIG2, PIN_DIG3, PIN_DIG4
+    };
+
+    gpio_clear(digit_pins[dev->current_digit]);
+    dev->current_digit += 1;
+    dev->current_digit = dev->current_digit % dev->params.digits;
+    gpio_set(digit_pins[dev->current_digit]);
+
+    DEBUG("[INFO] On display %d\n", dev->current_digit);
+
+    _set_pin_value(dev);
+}
+
+int digit7seg_init(digit7seg_t *dev, const digit7seg_params_t *params)
+{
+    dev->params = *params;
+    dev->current_digit = 0;
+    dev->value = 0;
+
+    if (dev->params.digits <= 0 || dev->params.digits > DIGIT7SEG_MAX_DIGITS) {
+        DEBUG("[Error] Invalid number of digit.\n");
+        return -DIGIT7SEG_ERR_DIGITS;
+    }
+
+    const gpio_t pins[] =
+    {
+        PIN_A, PIN_B, PIN_C, PIN_D, PIN_E, PIN_F, PIN_G, PIN_DP,
+        PIN_DIG1, PIN_DIG2, PIN_DIG3, PIN_DIG4
+    };
+
+    const int pin_errs[] =
+    {
+        DIGIT7SEG_ERR_A_GPIO, DIGIT7SEG_ERR_B_GPIO, DIGIT7SEG_ERR_C_GPIO, DIGIT7SEG_ERR_D_GPIO,
+        DIGIT7SEG_ERR_E_GPIO, DIGIT7SEG_ERR_F_GPIO, DIGIT7SEG_ERR_G_GPIO, DIGIT7SEG_ERR_DP_GPIO,
+        DIGIT7SEG_ERR_DIG1_GPIO, DIGIT7SEG_ERR_DIG2_GPIO, DIGIT7SEG_ERR_DIG3_GPIO,
+        DIGIT7SEG_ERR_DIG4_GPIO
+    };
+
+    for (int i = 0; i < NB_PIN; i++) {
+        if (!gpio_is_valid(pins[i])) {
```

```suggestion
        assert(gpio_is_valid(pins[i]);
```
If the config is wrong, fail loud and clear.
This is not something that changes at run-time.